rust/osm2lanes/src/transform/tags_to_lanes/bus.rs

```diff
@@ -55,84 +70,101 @@ pub(super) fn bus(
-            .set_bus(locale)?;
-        road.backward_outside_mut()
-            .ok_or_else(|| RoadError::unsupported_str("no backward lanes for busway"))?
-            .set_bus(locale)?;
-        if tags.is("oneway", "yes") || tags.is("oneway:bus", "yes") {
-            return Err(RoadMsg::ambiguous_str("busway:both=lane for oneway roads").into());
+        if tags.is(BUSWAY, "opposite_lane") {
+            busway.backward_direction = Infer::Direct(Some(Direction::Backward));
         }
-    }
-    if tags.is(BUSWAY + locale.driving_side.tag(), "lane") {
-        road.forward_outside_mut()
-            .ok_or_else(|| RoadError::unsupported_str("no forward lanes for busway"))?
-            .set_bus(locale)?;
-    }
-    if tags.is(BUSWAY + locale.driving_side.tag(), "opposite_lane") {
-        return Err(
-            RoadMsg::ambiguous_tag(BUSWAY + locale.driving_side.tag(), "opposite_lane").into(),
-        );
-    }
-    if tags.is(BUSWAY + locale.driving_side.opposite().tag(), "lane") {
-        if tags.is("oneway", "yes") || tags.is("oneway:bus", "yes") {
-            road.forward_inside_mut()
-                .ok_or_else(|| RoadError::unsupported_str("no forward lanes for busway"))?
-                .set_bus(locale)?;
-        } else {
-            return Err(
-                RoadMsg::ambiguous_str("busway:BACKWARD=lane for bidirectional roads").into(),
+        if tags.is(BUSWAY + "both", "lane") {
+            busway.forward_direction = Infer::Direct(Some(Direction::Forward));
+            busway.backward_direction = Infer::Direct(Some(Direction::Backward));
+            if tags.is("oneway", "yes") || tags.is("oneway:bus", "yes") {
+                warnings.push(RoadMsg::ambiguous_str("busway:both=lane for oneway roads").into());
+            }
+        }
+        if tags.is(BUSWAY + locale.driving_side.tag(), "lane") {
+            busway.forward_direction = Infer::Direct(Some(Direction::Forward));
+        }
+        if tags.is(BUSWAY + locale.driving_side.tag(), "opposite_lane") {
+            warnings.push(
+                RoadMsg::ambiguous_tag(BUSWAY + locale.driving_side.tag(), "opposite_lane").into(),
             );
         }
+        if tags.is(BUSWAY + locale.driving_side.opposite().tag(), "lane") {
+            if tags.is("oneway", "yes") || tags.is("oneway:bus", "yes") {
+                busway.forward_direction = Infer::Direct(Some(Direction::Forward));
+            } else {
+                warnings.push(
+                    RoadMsg::ambiguous_str("busway:BACKWARD=lane for bidirectional roads").into(),
+                );
+            }
+        }
+        if tags.is(
+            BUSWAY + locale.driving_side.opposite().tag(),
+            "opposite_lane",
+        ) {
+            if tags.is("oneway", "yes") || tags.is("oneway:bus", "yes") {
+                // TODO: does it make sense to have a backward lane on the forward_side????
+                busway.forward_direction = Infer::Direct(Some(Direction::Backward));
```

Comment:
aha, this is the edge case.
So in australia (LHT) given a oneway `busway:right=opposite_lane`

https://wiki.openstreetmap.org/wiki/Key:busway
> Bus lanes in both directions, on left and right sides of a one-way road: [busway]():[left](https://wiki.openstreetmap.org/wiki/Key:left)=opposite_lane + [busway]():[right](https://wiki.openstreetmap.org/wiki/Key:right)=lane

I would expect a bus lane, as the furthest right lane, going in the backward direction (i.e. contraflow).

I struggle with what `busway.forward_direction = Direction::Backward`  means.

Reply:
I guess you are working around the fact that there is no backward_direction lanes to turn into bus lanes?

Reply:
This is from the existing logic. I never did understand it, perhaps it was working around oneway only having forward lanes?

It is probably right to say `backward_direction = Backward` in this case, but that depends on the interpretation of the scheme. I'd say so, and this isn't covered by tests, so I'll make the change. In the context of a `oneway=yes` road "opposite" must mean "backward": 

> Bus lanes in both directions, on left and right sides of a one-way road:
busway:left=opposite_lane + busway:right=lane [RHT]
https://wiki.openstreetmap.org/wiki/Key:busway#Examples

`busway.forward_direction` is better named "forward side direction", so `busway.forward_side_direction = Direction::Backward` is how we describe what the code was doing: `road.forward_inside_mut().direction = Infer::Direct(Direction::Backward)` (though my impl puts the `Direction::Backward` lane on the outside).